src/hotspot/share/cds/heapShared.cpp

```diff
@@ -1554,7 +1554,6 @@ void HeapShared::archive_object_subgraphs(ArchivableStaticFieldInfo fields[],
   //     by any of these static fields.
   //     At runtime, these classes are initialized before X's archived fields
   //     are restored by HeapShared::initialize_from_archived_subgraph().
-  int i;
```

Comment:
This cleanup seems unrelated to the fix but I think it's fine to include it.

Reply:
This is necessary for the Windows build, which would fail because the `i` variable declared on this line is never used.